File: ChatGPT/Java/Facade/round3/source/Car.java
package ChatGPT.Java.Facade.round3.source;

public class Car {
     private Engine engine;
     private Lights lights;
     private Steering steering;
     private Brakes brakes;
 
     public Car() {
         this.engine = new Engine();
         this.lights = new Lights();
         this.steering = new Steering();
         this.brakes = new Brakes();
     }
 
     public void startCar() {
         engine.start();
         lights.turnOn();
         System.out.println("Car started.");
     }
 
     public void stopCar() {
         lights.turnOff();
         engine.stop();
         System.out.println("Car stopped.");
     }
 
     public void turnLeft() {
         steering.turnLeft();
     }
 
     public void turnRight() {
         steering.turnRight();
     }
 
     public void applyBrakes() {
         brakes.apply();
     }
 
     public void releaseBrakes() {
         brakes.release();
     }
 }
 

File: ChatGPT/Python/Facade/round1/source/car.py
# car.py
from engine import Engine
from lights import Lights
from steering import Steering
from brakes import Brakes

class CarFacade:
    def __init__(self):
        self.engine = Engine()
        self.lights = Lights()
        self.steering = Steering()
        self.brakes = Brakes()

    def start_car(self):
        self.engine.start()
        self.lights.turn_on()

    def stop_car(self):
        self.lights.turn_off()
        self.engine.stop()

    def turn_left(self):
        self.steering.turn_left()

    def turn_right(self):
        self.steering.turn_right()

    def apply_brakes(self):
        self.brakes.apply()

    def release_brakes(self):
        self.brakes.release()
